### PR TITLE
fix Issue 17246 - [REG2.053] Extra destructor call.

### PR DESCRIPTION
**Pull request by WalterBright**
This is a critical bug fix, as it can corrupt memory.

### CODE REVIEW

**Comment by mathias-lang-sociomantic** [comment]
> This is a critical bug fix, as it can corrupt memory.

Then shouldn't it target stable ?

**Review by andralex — APPROVED** [review]
cc @MartinNowak

**Comment by mathias-lang-sociomantic** [comment]
@andralex : Why did you merge it in master despite the review ?

**Comment by andralex** [comment]
@mathias-lang-sociomantic leaving the decision to @MartinNowak the build czar

**Comment by mathias-lang-sociomantic** [comment]
And if it goes into stable, Martin has to cherry-pick it. We never merge master into stable, always the other way around. So if there's a reasonable doubt about a P.R. needing to go to stable, it shouldn't be merged right away like it just happened.

**Comment by andralex** [comment]
Cool, @MartinNowak please advise on how you prefer to proceed.

File: src/ddmd/expression.d [hunk 1]
```diff
@@ -1483,16 +1483,20 @@ extern (C++) bool preFunctionParameters(Loc loc, Scope* sc, Expressions* exps)
  */
 extern (C++) Expression valueNoDtor(Expression e)
 {
-    if (e.op == TOKcall)
+    auto ex = e;
+    while (ex.op == TOKcomma)
```
Comment (stefan-koch-sociomantic) [inline src/ddmd/expression.d]: This is such a common thing it's time we factor it into a function isn't it ?

**Pull request merged**